.github/workflows/main.yml

```
@@ -0,0 +1,86 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+jobs:
+  ksc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: build-compiler
+        run: |
+          cd compiler
+          export GIT_COMMIT=$(git log -1 --format=%h)
+          export GIT_DATE_ISO=$(TZ=UTC git log -1 --date=iso-strict-local --format=%cd)
+          export GIT_DATE=$(TZ=UTC git log -1 --date=format-local:%Y%m%d.%H%M%S --format=%cd)
+          export KAITAI_STRUCT_VERSION=0.11-SNAPSHOT${GIT_DATE}.${GIT_COMMIT}
+
+          echo "KAITAI_STRUCT_VERSION=$KAITAI_STRUCT_VERSION"
+
+          sbt \
```

I'd also add this:

```suggestion
          cat /dev/null | sbt \
```

Normally, `sbt` runs in interactive mode, which means it may request user input if something's wrong. I saw this in the Travis build (https://app.travis-ci.com/github/kaitai-io/kaitai_struct/builds/264091863#L477-L481):

```
[error] Environment variable KAITAI_STRUCT_VERSION '0.10-SNAPSHOT20230626.194946.766c60ed' doesn't start with build.sbt VERSION '0.11-SNAPSHOT'
[warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

So basically the build had to time out itself (which on Travis happened after 10 minutes), because `sbt` was waiting on the user input.

Piping `/dev/null` to stdin should put `sbt` to non-interactive mode, so it should exit immediately if there's an error.

---

Good catch, I wasn't aware of that. Let me fix that.